dirty_cat/datasets/_fetching.py

```diff
@@ -61,16 +61,33 @@ class Features:
 
 @dataclass(unsafe_hash=True)
 class DatasetAll:
+    """
+    Represents a dataset and its information.
+    With this state, the dataset is loaded in memory as a pandas DataFrame
+    (``X`` and ``y``).
+    """
+
     name: str
     description: str
+    source: str
+    target: str
     X: pd.DataFrame
     y: pd.Series
-    source: str
     path: Path
+    read_csv_kwargs: Dict[str, Any]
```

Comment:
You've explained really well below why `read_csv_kwargs` was added to `DatasetInfoOnly` but I don't see the utility for the user of adding it here. It multiplies information in this already complex class and is not used in this case.

Reply:
I added a little clarification for why we pass this information, even though the dataset is loaded in memory.
Mainly, it's because we want to have this information in case the dataframe has to be read from the file again.